core-data/reducer.js

```diff
@@ -43,68 +48,76 @@ export function terms( state = {}, action ) {
 }
 
 /**
- * Reducer managing media state. Keyed by id.
+ * Reducer managing theme supports data.
  *
  * @param {Object} state  Current state.
  * @param {Object} action Dispatched action.
  *
  * @return {Object} Updated state.
  */
-export function media( state = {}, action ) {
+export function themeSupports( state = {}, action ) {
 	switch ( action.type ) {
-		case 'RECEIVE_MEDIA':
+		case 'RECEIVE_THEME_SUPPORTS':
 			return {
 				...state,
-				...keyBy( action.media, 'id' ),
+				...action.themeSupports,
 			};
 	}
 
 	return state;
 }
 
 /**
- * Reducer managing post types state. Keyed by slug.
+ * Higher Order Reducer for a given entity config. It supports:
  *
- * @param {Object} state  Current state.
- * @param {Object} action Dispatched action.
+ *  - Fetching a record by primariy key
  *
- * @return {Object} Updated state.
+ * @param {Object} entityConfig  Entity config.
+ *
+ * @return {Function} Reducer.
  */
-export function postTypes( state = {}, action ) {
-	switch ( action.type ) {
-		case 'RECEIVE_POST_TYPES':
-			return {
-				...state,
-				...keyBy( action.postTypes, 'slug' ),
-			};
-	}
+function entity( entityConfig ) {
+	return ( state = { byKey: {} }, action ) => {
+		if (
+			! action.name ||
+			! action.kind ||
+			action.name !== entityConfig.name ||
+			action.kind !== entityConfig.kind
+		) {
+			return state;
+		}
 
-	return state;
+		const key = entityConfig.key || 'id';
```

Comment:
Minor (Perf): We don't need to derive this on every dispatch. It can be in the surrounding function scope.